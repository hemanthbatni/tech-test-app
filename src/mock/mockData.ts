import { Product } from "../store/productsSlice";

interface ProductsResponse {
  data: {
    search: {
      total: number;
      products: Product[];
    };
  };
}

export const productsMock: ProductsResponse = {
  data: {
    search: {
      total: 4,
      products: [
        {
          id: "product-id-1",
          name: "Prohibition Liquor Company",
          description: "Prohibition Christmas 500ml",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: "sale",
          wasPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 200,
            },
            pointsPrice: {
              amount: 25000,
            },
          },
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 119,
            },
            pointsPrice: {
              amount: 15000,
            },
          },
        },
        {
          id: "product-id-2",
          name: "Houraisen",
          description: "Houraisen Beshi Tokubetsu Junmai Sake 720ml",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: "sale",
          wasPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 134,
            },
            pointsPrice: {
              amount: 23470,
            },
          },
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 129.99,
            },
            pointsPrice: {
              amount: 19900,
            },
          },
        },
        {
          id: "product-id-3",
          name: "Royal Salute",
          description:
            "Royal Salute 21 Year Old Blended Scotch Whisky Miami Polo Edition 700ml (Gift Box)",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: null,
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 399,
            },
            pointsPrice: {
              amount: 79800,
            },
          },
        },
        {
          id: "product-id-4",
          name: "Aberlour",
          description:
            "Aberlour 16 Year Old Double Cask Single Malt Scotch Whisky 700ml (Gift Tube)",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: null,
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 199.9,
            },
            pointsPrice: {
              amount: 39980,
            },
          },
        },
        {
          id: "product-id-5",
          name: "Prohibition Liquor Company",
          description: "Prohibition Christmas 500ml",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: "sale",
          wasPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 200,
            },
            pointsPrice: {
              amount: 25000,
            },
          },
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 119,
            },
            pointsPrice: {
              amount: 15000,
            },
          },
        },
        {
          id: "product-id-6",
          name: "Houraisen",
          description: "Houraisen Beshi Tokubetsu Junmai Sake 720ml",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: "sale",
          wasPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 134,
            },
            pointsPrice: {
              amount: 23470,
            },
          },
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 129.99,
            },
            pointsPrice: {
              amount: 19900,
            },
          },
        },
        {
          id: "product-id-7",
          name: "Royal Salute",
          description:
            "Royal Salute 21 Year Old Blended Scotch Whisky Miami Polo Edition 700ml (Gift Box)",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: null,
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 399,
            },
            pointsPrice: {
              amount: 79800,
            },
          },
        },
        {
          id: "product-id-8",
          name: "Aberlour",
          description:
            "Aberlour 16 Year Old Double Cask Single Malt Scotch Whisky 700ml (Gift Tube)",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: null,
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 199.9,
            },
            pointsPrice: {
              amount: 39980,
            },
          },
        },
      ],
    },
  },
};
