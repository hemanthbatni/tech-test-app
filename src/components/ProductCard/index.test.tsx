import { createEvent, fireEvent, render, screen } from "@testing-library/react";
import { ProductCard } from ".";
import { productsMock } from "../../data/mockData";

jest.spyOn(global.console, "error").mockImplementation();

it("renders component", () => {
  const product = productsMock.data.search.products[0];
  const { container, getByTestId } = render(<ProductCard product={product} />);

  expect(container).toMatchSnapshot();
  expect(getByTestId("show-product-details")).toBeInTheDocument();
});

it("renders correct on sale product details", () => {
  const product = productsMock.data.search.products[0];
  const { container, getByTestId } = render(<ProductCard product={product} />);

  expect(getByTestId("show-product-details")).toBeInTheDocument();
  expect(getByTestId("product-tag").innerHTML).toEqual("sale");
  expect(getByTestId("product-name").innerHTML).toEqual(
    "Prohibition Liquor Company"
  );
  expect(getByTestId("product-was-cash-price").innerHTML).toEqual("$200");
  expect(getByTestId("product-current-cash-price").innerHTML).toEqual("$119");
  expect(getByTestId("product-was-points-price").innerHTML).toEqual(
    "25000 PTS"
  );
  expect(getByTestId("product-current-points-price").innerHTML).toEqual(
    "15000 PTS"
  );
  expect(getByTestId("add-to-cart-button")).toBeInTheDocument();
  expect(getByTestId("add-to-cart-button")).toBeEnabled();
});

it("renders correct normal product details", () => {
  const product = productsMock.data.search.products[2];
  const { container, getByTestId } = render(<ProductCard product={product} />);

  expect(getByTestId("show-product-details")).toBeInTheDocument();
  expect(getByTestId("product-was-cash-price").innerHTML).toEqual(" ");
  expect(getByTestId("product-was-points-price").innerHTML).toEqual(" ");
});
