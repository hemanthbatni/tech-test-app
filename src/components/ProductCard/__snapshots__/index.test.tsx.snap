// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<div>
  <div
    class="css-fm7ljw"
  >
    <a
      aria-label="click to view product details"
      class="css-1h5x3dy"
      data-testid="show-product-details"
    >
      <div
        class="css-lx9y6m"
      >
        <span
          class="css-1ijmt3n"
          data-testid="product-tag"
        >
          sale
        </span>
      </div>
      <div
        class="css-1q43t41"
      >
        <h4
          class="css-39fqji"
          data-testid="product-name"
        >
          Prohibition Liquor Company
        </h4>
        <div
          class="css-10dpuw1"
        >
          Prohibition Christmas 500ml
        </div>
      </div>
      <div
        class="css-rjmmzf"
      >
        <img
          class="css-orsq6l"
          src="https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp"
        />
      </div>
      <div
        class="css-1q43t41"
      >
        <div
          aria-hidden="false"
          class="css-1fwy7tj"
          data-testid="product-was-cash-price"
        >
          $200
        </div>
        <div
          class="css-14o6v6q"
          data-testid="product-current-cash-price"
        >
          $
          119
        </div>
      </div>
      <div
        class="css-bahvbi"
      >
        <div
          class="css-n6vmit"
        >
          <div
            class="css-53np2w"
          >
            <span>
              Or
            </span>
            <div
              class="css-10x14qo"
              data-testid="product-current-points-price"
            >
              15000
               PTS
            </div>
          </div>
          <div
            aria-hidden="false"
            class="css-1fwy7tj"
            data-testid="product-was-points-price"
          >
            25000 PTS
          </div>
        </div>
        <div>
          <button
            aria-label="click to add product to cart"
            class="css-1anwsz4"
            data-testid="add-to-cart-button"
          >
            <div>
              <p>
                ADD
              </p>
            </div>
          </button>
        </div>
      </div>
    </a>
  </div>
</div>
`;
